src/gt4py/gtc/gtcpp/gtcpp_codegen.py

```diff
@@ -69,6 +69,14 @@ class GTCppCodegen(codegen.TemplatedGenerator):
 
     Cast = as_fmt("static_cast<{dtype}>({expr})")
 
+    def visit_BuiltInLiteral(self, builtin: BuiltInLiteral, **kwargs):
+        if builtin == BuiltInLiteral.TRUE:
+            return "true"
+        elif builtin == BuiltInLiteral.FALSE:
+            return "false"
+        else:
+            assert False
```

Comment:
```suggestion
        if builtin == BuiltInLiteral.TRUE:
            return "true"
        elif builtin == BuiltInLiteral.FALSE:
            return "false"
        assert False
        return None
```

However, have you considered throwing an exception? There is a flag in the python interpreter which optimizes asserts away, in which case this unexpectedly returns None (made explicit above, but happens also without that line).

Reply:
done.